### PR TITLE
Update serialize flag for bevy_ecs

### PR DESCRIPTION
**Pull request by cBournhonesque**
# Objective

There were some issues with the `serialize` feature:
- `bevy_app` had a `serialize` feature and a dependency on `serde` even there is no usage of serde at all inside `bevy_app`
- the `bevy_app/serialize` feature enabled `bevy_ecs/serde`, which is strange
- `bevy_internal/serialize` did not enable `bevy_app/serialize` so there was no way of serializing an Entity in bevy 0.14

## Solution

- Remove `serde` and `bevy_app/serialize` 
- Add a `serialize` flag on `bevy_ecs` that enables `serde`
- ` bevy_internal/serialize` now enables `bevy_ecs/serialize`

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
> bevy_internal/serialize did not enable bevy_app/serialize so there was no way of serializing an Entity in bevy 0.14

Because features are additive, you can actually get this by pulling a direct dependency on `bevy_ecs` and enabling the feature there. But this should still be fixed!

File: crates/bevy_app/Cargo.toml [hunk 1]
```diff
@@ -13,7 +13,6 @@ trace = []
 bevy_debug_stepping = []
 default = ["bevy_reflect"]
 bevy_reflect = ["dep:bevy_reflect", "bevy_ecs/bevy_reflect"]
-serialize = ["bevy_ecs/serde"]
```
Comment (alice-i-cecile) [inline crates/bevy_app/Cargo.toml:16]: Does `bevy_app` really not have any other serialization functionality?

**Comment by alice-i-cecile** [comment]
CI seems very confused about your PR: I don't understand either the failures or the warning about missing examples.

File: crates/bevy_app/Cargo.toml (hunk 1 above)
Comment (cBournhonesque) [reply crates/bevy_app/Cargo.toml:16]: Ctrl+F showed 0 hits!

**Comment by cBournhonesque** [comment]
@alice-i-cecile should be good now :)

**Review by alice-i-cecile — APPROVED** [review]

**Review by NiseVoid — APPROVED** [review]

**Comment by knutsoned** [comment]
Just tested this with leafwing-input-manager and rc.4. The bevy_ecs serde flag hack now causes an error. Removing it seems to fix the build error.

**Pull request merged**